### PR TITLE
Log

### PR DESCRIPTION
**Pull request by otaviobp**
Adding options to change log level in command line and fix a bug in logging

### CODE REVIEW

File: src/log.c [hunk 1]
```diff
@@ -121,6 +121,9 @@ int log_internal(int level, int error, const char *file, int line, const char *f
 {
     va_list ap;
     int r;
+    int _level = (level);
+    if (log_get_max_level() < LOG_PRI(_level))
+        return -error;
```
Comment (lucasdemarchi) [inline src/log.c]: This is actually done by log_full_errno() that should be the interface being used.

File: src/main.cpp [hunk 2]
```diff
@@ -167,6 +189,20 @@ static int parse_argv(int argc, char *argv[], struct options *opt)
...
+                help(stderr);
+                return -EINVAL;
+            }
+            log_set_max_level(lvl);
+            break;
+        }
+        case 'v': {
+            log_set_max_level(LOG_DEBUG);
+            break;
+        }
```
Comment (lucasdemarchi) [inline src/main.cpp]: /me confused... what exactly is not working?

File: src/main.cpp [hunk 3]
```diff
@@ -144,17 +149,34 @@ static int parse_conf_files(ConfFile &conf, struct options *opt)
     return ret;
 }
 
+static int log_level_from_str(const char *str)
```
Comment (lucasdemarchi) [inline src/main.cpp]: this can be part of log.h/log.c

File: src/main.cpp [hunk 4]
```diff
@@ -38,14 +39,18 @@ struct options {
...
+        "%s [OPTIONS...]\n\n"
+        "  -c --conf-file                   .conf file with configurations for "
+        "camera-streaming-daemon.\n"
+        "  -d --conf-dir <dir>              Directory where to look for .conf files overriding\n"
+        "                                   default conf file.\n"
+        "  -g --debug-log-level <level>     Set debug log level. Levels are\n"
+        "                                   <error|warning|notice|info|debug>\n"
+        "  -v --verbose                     Verbose. Same as --debug-log-level=debug\n"
+        "  -h --help                        Print this message\n",
+        program_invocation_short_name);
```
Comment (lucasdemarchi) [inline src/main.cpp]: Unrelated changes should be avoided. And following the same coding style as in our other projects would help. Also, there's room for improvement in the script to check coding style that was started on mavlink-router.

File: src/main.cpp (hunk 3 above)
Comment (otaviobp) [inline src/main.cpp]: I used same approach from mavlink-routerd, that keeps this on main.c

File: src/main.cpp (hunk 2 above)
Comment (otaviobp) [inline src/main.cpp]: This commit is adding -v and -g parameters to csd. It wasn't handled before this patch

File: src/log.c (hunk 1 above)
Comment (otaviobp) [inline src/log.c]: There is no log_full_errno in csd. Maybe we could use same version from mavlink-router, but that is not the case right now.

File: src/main.cpp (hunk 4 above)
Comment (otaviobp) [inline src/main.cpp]: This is not unrelated changed. And I didn't see that the check-coding has messed up with code. I'll fix the style

**Comment by otaviobp** [comment]
merged using log imported from mavlink-router.
